File: 189-rotate-array/189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        if(k%n==0){
            return;
        }
        int v=k%n;
        vector<int> temp;
        for(int i=v;i<n;i++){
            temp.push_back(nums.at(i));
        }
        for(int i=0,j=0;i<n;i++){
            if(i<v){
                nums.at((i+v)%n)=nums.at(i);
            }
            else {
                nums.at((i+v)%n)=temp.at(j);
                j++;
            }
        }
    }
};

File: 189-rotate-array/189-rotate-array.java
class Solution {
    private void arrRev(int[] arr,int st,int end){
		for(int i=st,j=end;i<j;i++,j--) {
			arr[i]=arr[i]^arr[j];
			arr[j]=arr[j]^arr[i];
			arr[i]=arr[i]^arr[j];
		}
	}
    public void rotate(int[] nums, int k) {
        int arL=nums.length;
        k=k%arL;
        arrRev(nums,0,arL-k-1);
        arrRev(nums,arL-k,arL-1);
        arrRev(nums,0,arL-1);
    }
}